src/main/webapp/meal-page-script.js

```diff
@@ -39,21 +42,79 @@ function createElementByTag(text, tag) {
   return element;
 }
 
-/** Creates a map and adds it to the page. */
+let map;
+
 function createMap() {
-  const userLocation = new google.maps.LatLng(55.746514, 37.627022);
+  let userLocation = new google.maps.LatLng(55.746514, 37.627022);
   const mapOptions = {
-    zoom: 16,
-    center: userLocation
+    zoom: 14,
+    center: userLocation,
   };
-  const map = new google.maps.Map(document.getElementById("map"), mapOptions);
-  // TODO(grenlayk): implement real geolocation here
-  // TODO(grenlayk): change "You are here" marker on custom one
+  map = new google.maps.Map(document.getElementById("map"), mapOptions);
   const locationMarker = new google.maps.Marker({
     position: userLocation,
     map,
-    title: "You are here",
+    title: "You are here (probably)",
+    label: "U",
     animation: google.maps.Animation.DROP,
   });
+
+  
+  if (navigator.geolocation) {
+    navigator.geolocation.getCurrentPosition(
+      function (position) {
+        const location = new google.maps.LatLng(
+          position.coords.latitude,
+          position.coords.longitude
+        );
+        map.setCenter(location);
+        locationMarker.setPosition(location);
+        addRestaurants();
+      },
+      function () {
+        handleLocationError(true);
+        addRestaurants();
+      }
+    )
+  } else {
+    handleLocationError(false);
+    addRestaurants();
+  }
+}
+
+function addRestaurants() {
+  // Hard coded restaurants
+  const restaurants = [];
+  restaurants.push(changeLocation(map.getCenter(), 0.001, 0.001));
```

Comment:
Nit:
this is a Java-ish approach: define an empty list and then call `push` a few times.
In JS you'd normally do it in one line:
```
const restaurants = [changeLocation(map.getCenter(), 0.001, 0.001), changeLocation(map.getCenter(), -0.001, 0.001)];
```